core/services/run_queue.go

```diff
@@ -56,16 +57,22 @@ func (rq *runQueue) Start() error {
 
 // Stop closes all open worker channels.
 func (rq *runQueue) Stop() {
+	rq.workersMutex.Lock()
+	rq.stopRequested = true
+	rq.workersMutex.Unlock()
```

Comment:
use the pattern 'defer rq.workersMutex.Unlock()` ;)  its a common pattern and keeps the concept together

Reply:
Can't defer here @felder-cl . We _must_ unlock before line 63.

Reply:
That's another deadlock! (avoided)